props/hev.h

```diff
@@ -0,0 +1,440 @@
+#ifndef PROPS_HEV_H
+#define PROPS_HEV_H
+#define PROP_TYPE Hev
+
+// Half-Life Hazardous Environment Suit Prop
+//
+// How to use:
+// Power button:
+//   Long-click: on/off
+//   double-click: sound track on/off
+//   triple-click: next preset
+// AUX button:
+//   hold button to heal.
+//   single click: deactivate hazard
+//   double-click: armor readout
+//   triple-click: previous preset
+// Clash will cause varying damage based on strength
+//
+// Damage and Hazards are only active when prop is on.
+
+#include "prop_base.h"
+#include <cmath>
+
+EFFECT(health);
+EFFECT(armor);
+EFFECT(stun);
+EFFECT(death);
+EFFECT(armor_alarm);
+EFFECT(armor_compromised);
+EFFECT(major);
+EFFECT(minor);
+EFFECT(morphine);
+
+class Hev : public PROP_INHERIT_PREFIX PropBase {
+public:
+  Hev() : PropBase() {}
+  const char* name() override { return "Hev"; }
+
+  int health_ = 100;
+  int armor_ = 100;
+
+  enum Hazard {
+    HAZARD_NONE = 0,
+    HAZARD_BIO = 1,
+    HAZARD_RAD = 2,
+    HAZARD_BLO = 3,
+    HAZARD_CHE = 4,
+    HAZARD_HEA = 5,
+    HAZARD_SHO = 6,
+  };
+
+  Hazard current_hazard_ = HAZARD_NONE;
+  
+  void LowBatteryOff() override {
+    if (SFX_poweron) {
+      PropBase::LowBatteryOff();
+    }
+  }
+
+
+  // Calculate Physical Damage //////////////////////////////////////////////////
+  void DoDamage(int damage, bool quiet = false) {
+    int tens = health_ / 10;
+    if (armor_ >= damage) {
+      armor_ -= ceilf(damage * 0.80 / 2);
+      health_ -= ceilf(damage * 0.20);
+    } else if (armor_ > 0) {
+      int excess_physical = damage - armor_;
+      health_ -= excess_physical;
+      // Make armor compromised sound play before setting armor to 0
+      SaberBase::DoEffect(EFFECT_USER2, 0.0, 0);
+      PVLOG_NORMAL << "Armor Compromised!\n";
+      armor_ = 0;
+    } else {
+      health_ -= damage;
+    }
+
+    // Enforce minimum values
+    if (health_ < 0) health_ = 0;
+    if (armor_ < 0) armor_ = 0;
+
+    // Damage
+    if (!quiet) SaberBase::DoEffect(EFFECT_STUN, 0.0, damage);
+    
+    // Death Sound
+    if (health_ == 0) {
+      SaberBase::DoEffect(EFFECT_EMPTY, 0.0, 100);
+      return;
+    }
+    
+    // Health Alert - only plays when health enters a new multiple of 10
+    int new_tens = health_ / 10;
+    if (tens != new_tens) {
+      SaberBase::DoEffect(EFFECT_USER1, 0.0, 0);
+    }
+
+    // Print health and armor
+    PVLOG_NORMAL << "Health: " << health_ << " / ";
+    PVLOG_NORMAL << "Armor: " << armor_ << "\n";
+  }
+
+
+  // Armor Readout /////////////////////////////////////////////////////////////
+  void armor_readout() {
+    PVLOG_NORMAL << "Current Armor: " << armor_ << "\n"; // Debug logging
+    SFX_armor.SelectFloat(armor_ / 100.0);
+    hybrid_font.PlayCommon(&SFX_armor);
+  }
+
+
+  // Clashes ///////////////////////////////////////////////////////////////////
+  uint32_t last_clash_time_ = 0;
```

Comment:
Since you're doing a lot with timers, it might be helpful to have a helper class, something like:

```
struct HEVTimer {
   bool reset() { active_ = false; }
   void start() { active_ = true; start_ = millis(); }
   bool check(uint32_t timeout) {
     if (!active_) return false;
     if (millis() - start_ > timeout) return true;
     active_ = false;
     return false;
   }
   uint32_t start_;
   bool active_ = false;
};
```